waku/v2/node/rest/test_debug_api_serdes.nim

```diff
@@ -0,0 +1,38 @@
+import std/typetraits
+import chronicles,
+  unittest,
+  stew/[results, byteutils],
+  json_serialization
+import "."/[json_serdes, debug_api]
+
+
+suite "Debug API - serialization":
+
+  suite "DebugWakuInfo - decode":
+    test "optional field is not provided":
+      # Given
+      let jsonString = JsonString("""{ "listenAddresses":["123"] }""")
+
+      # When
+      let res = decodeJsonString(DebugWakuInfo, jsonString, requireAllFields = true)
+
+      # Then
+      require(res.isOk)
```

Comment:
Really like this logical structuring of unit tests. Good job!

Reply:
This comes from my time as QA, structuring tests like this helped me a lot to focus on the _setup-stimulus-assertion_ way of thinking; and I find it very helpful in order to apply the SOLID principles because it helps you to think that a "unit" should do one and only one thing.

From: https://www.browserstack.com/guide/unit-testing-best-practices

> Generally, unit tests comprise three phases:
> 
> * Initialize the application module to be tested AKA the system under test
> * Apply stimulus to the system under tests
> * Observe resultant behavior

And, given that you were a JVM guy 😁 I like very much the way Groovy's Spock models the test cases: https://www.baeldung.com/groovy-spock#2-blocks